BIDSTools/Process_BIDS_DATA.py

```diff
@@ -0,0 +1,519 @@
+
+    The following directory structure will be created:
+        output_dir/sub-01/ses-01/func/
+    """
+    list_experiments_already_processed = []
+    current_dir = output_dir
+
+    # Retrieve attributes from the experiment object
+    subject_id = experiment.get_attribute("Subject ID")
+    data_type = experiment.get_attribute("Data type")
+    session_number = experiment.get_attribute("Session Number")
+
+    # Create the subject directory
+    current_dir = generate_subdir(current_dir, subject_id)
+
+    # If the data type requires a session directory, create it
+    if check_datatype_has_session(data_type):
+        current_dir = generate_session_dir(current_dir, session_number)
+
+    # Create the data type directory
+    current_dir = generate_datatype_dir(current_dir, data_type)
+
+    # Append the processed experiment to the list
+    list_experiments_already_processed.append(experiment)
+
+    return current_dir, list_experiments_already_processed
+
+
+def add_new_experiment_to_tsv(file_path, experiment):
+    """
+    Adds a new experiment (a new row from the metadata CSV file) to the TSV file.
+
+    Parameters
+    ----------
+    file_path : str
+        The path to the TSV file to which the new row (experiment) will be added.
+    experiment : Experiment
+        An object from the Experiment class representing a row of the metadata file.
+
+    Returns
+    -------
+    None
+        The function modifies the TSV file in place by adding a new row.
+    """
+    # Convert the experiment object to a dictionary
+    exp_dict = experiment.to_dict()
+
+    # Read the existing header from the file if it exists
+    if os.path.isfile(file_path):
+        with open(file_path, 'r') as f:
+            # Read the first line to get the header
+            header_line = f.readline()
+            # Split the header line into field names
+            existing_fieldnames = header_line.strip().split('\t')
+    else:
+        # If the file doesn't exist, initialize an empty list of field names
+        existing_fieldnames = []
+
+    # Open the file in append mode
+    with open(file_path, 'a', newline='') as f:
+        # If the file is new and empty, initialize with all keys from exp_dict
+        if not existing_fieldnames:
+            print("its have a problem with this file {}".format(file_path))
+            pass
+            existing_fieldnames = exp_dict.keys()
+            writer = csv.DictWriter(f, delimiter='\t', fieldnames=existing_fieldnames)
+            writer.writeheader()
+        else:
+            # Filter the experiment dictionary to include only existing fields
+            filtered_exp_dict = {key: exp_dict.get(key, '') for key in existing_fieldnames}
+            writer = csv.DictWriter(f, delimiter='\t', fieldnames=existing_fieldnames)
+
+        # Write the new row (experiment)
+        writer.writerow(filtered_exp_dict if existing_fieldnames else exp_dict)
+
+
+def add_new_experiment_to_json(file_path, experiment):
```

Comment:
the docstring is missing

Reply:
also, this function does not seem to be used in the main algorithm, no? is this normal?